tests/integration/test_charm.py

```diff
@@ -1,32 +1,77 @@
 
 import pytest
-import yaml
+import requests
+from conftest import deploy  # noqa: F401, pylint: disable=W0611
+from helpers import (
+    APP_NAME_AIRBYTE_SERVER,
+    check_airbyte_job_status,
+    create_airbyte_connection,
+    create_airbyte_destination,
+    create_airbyte_source,
+    get_airbyte_workspace_id,
+    get_db_password,
+    get_unit_url,
+    trigger_airbyte_connection,
+)
 from pytest_operator.plugin import OpsTest
 
 logger = logging.getLogger(__name__)
 
-METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
-APP_NAME = METADATA["name"]
-
 
 @pytest.mark.abort_on_fail
-async def test_build_and_deploy(ops_test: OpsTest):
-    """Build the charm-under-test and deploy it together with related charms.
-
-    Assert on the unit status before any relations/configurations take place.
-    """
-    # Build and deploy charm from local source folder
-    charm = await ops_test.build_charm(".")
-    resources = {"httpbin-image": METADATA["resources"]["httpbin-image"]["upstream-source"]}
-
-    # Deploy the charm and wait for active/idle status
-    await asyncio.gather(
-        ops_test.model.deploy(charm, resources=resources, application_name=APP_NAME),
-        ops_test.model.wait_for_idle(apps=[APP_NAME], status="active", raise_on_blocked=True, timeout=1000),
-    )
+@pytest.mark.usefixtures("deploy")
+class TestDeployment:
+    """Integration tests for charm."""
+
+    async def test_deployment(self, ops_test: OpsTest):
+        url = await get_unit_url(ops_test, application=APP_NAME_AIRBYTE_SERVER, unit=0, port=8001)
+        logger.info("curling app address: %s", url)
+
+        response = requests.get(f"{url}/api/v1/health", timeout=300)
+
+        assert response.status_code == 200
+        assert response.json().get("available")
+
+    async def test_sync_job(self, ops_test: OpsTest):
+        # Create connection
+        api_url = await get_unit_url(ops_test, application=APP_NAME_AIRBYTE_SERVER, unit=0, port=8006)
+        logger.info("curling app address: %s", api_url)
+        workspace_id = get_airbyte_workspace_id(api_url)
+        db_password = await get_db_password(ops_test)
+
+        # Create Source
+        source_id = create_airbyte_source(api_url, workspace_id)
+
+        # Create destination
+        destination_id = create_airbyte_destination(api_url, ops_test.model.name, workspace_id, db_password)
+
+        # Create connection
+        connection_id = create_airbyte_connection(api_url, source_id, destination_id)
+
+        # Trigger sync job
+        job_id = trigger_airbyte_connection(api_url, connection_id)
+
+        # Wait until job is successful
+        job_successful = False
+        for i in range(30):
+            logger.info(f"attempt {i} to get job status")
+            status = check_airbyte_job_status(api_url, job_id)
+
+            if status == "failed":
+                break
+
+            if status == "succeeded":
+                logger.info(f"attempt {i}: job successful!")
```

Comment:
is this storing pikachu in the db or smth? :smile:

Reply:
Yup :D